yoyodyne/train.py

```diff
@@ -36,49 +36,65 @@ def _get_logger(experiment: str, model_dir: str, log_wandb: bool) -> List:
 
 
 def _get_callbacks(
-    save_top_k: int, patience: Optional[int] = None, save_best: bool = True
+    patience: Optional[int] = None,
+    top_k: int = defaults.TOP_K,
+    val_mode: str = defaults.VAL_MODE,
 ) -> List:
     """Creates the callbacks.
 
     We will reach into the callback metrics list to picks ckp_callback to find
     the best checkpoint path.
 
     Args:
-        save_top_k (int).
         patience (int, optional).
-        save_best (bool). If `True`, save the best checkpoint(s) using the
-            validation metric.
+        val_mode (str, optional).
+        top_k (int, optional)
+            saved each epoch.
 
     Returns:
         List: callbacks.
+
+    Raises:
+        Error: Unknown save mode.
     """
-    if save_best:
-        checkpoint_callback = callbacks.ModelCheckpoint(
-            save_top_k=save_top_k,
-            monitor="val_accuracy",
-            mode="max",
-            filename="model-{epoch:03d}-{val_accuracy:.3f}",
-        )
-    else:
-        checkpoint_callback = callbacks.ModelCheckpoint(
-            save_top_k=save_top_k,
-            every_n_epochs=1,
-            save_on_train_epoch_end=True,
-            filename="model-{epoch:03d}",
-        )
     trainer_callbacks = [
-        checkpoint_callback,
         callbacks.LearningRateMonitor(logging_interval="epoch"),
         callbacks.TQDMProgressBar(),
     ]
+    # Parses validation mode.
+    if val_mode == "accuracy":
```

Comment:
Up to you, but I think these conditionals could contain code just to set `mode`, `monitor`, and `filename`, and then the code block that appends the callback can just be written once, using these 3 variables.

Reply:
I tried that in an earlier rendition---and it worked, I just wasn't sure if I liked it. I'll restore it.